### PR TITLE
Workaround for NIC hotplugging with chrooted qemu and QA suite fix for ACPI/hotplugging

### PR DESCRIPTION
**Pull request by rbott**
This PR addresses two problems:

* hotplugging does not work when ACPI is disabled - as this might be the case during the QA runs (e.g. due to alternating through hvparams), the QA suite now only runs hotplug tests when ACPI is enabled for the guest
* while fixing the above I stumbled on another problem: when QEMU runs chrooted, one can not hotplug a NIC because qemu tries to load a PXE rom file which is not present in the chroot (`failed to find romfile "$depends_on_the_driver.rom"` error message). When chrooting is enabled, QMPs `device_add` will now be instructed to not load a rom file from disk

I guess the latter is more of a workaround. Either Ganeti should setup the chroot environment correctly (e.g. provide required rom files inside the chroot) or have the actual hotplug command fail early with a note that hotplugging nics is not possible while running qemu in a chroot. OTOH I am not sure how relevant PXE roms are when NICs are hot-added to already running systems.

### CODE REVIEW

File: lib/hypervisor/hv_kvm/__init__.py
Comment (saschalucas) [inline lib/hypervisor/hv_kvm/__init__.py:2300]: this is longer than 80c

File: lib/hypervisor/hv_kvm/monitor.py
Comment (saschalucas) [inline lib/hypervisor/hv_kvm/monitor.py:505]: Did you discover, that an empty romfile is needed, too, when `vhost_net` is enabled?

**Review by saschalucas — APPROVED** [review]
Thanks @rbott. Nice work, also solving #1621. I agree, that passing a ROM for hotpluged NIC seems not to make sense during runtime, just for startup/boot. However we are fine here to leave the ROM empty only when chrooted.

Please fix the 80c issue and double check, why the `romfile` is empty, too, for `vhost_net`. Other than that, LGTM. 

Generally the ROM seems useless, as long as there is no intention to boot from PXE? Maybe Qemu (or Ganeti) should not load a ROM, unless explicitly requested (i.e. by `boot_order`)?

**Comment by saschalucas** [comment]
Added label for Backporting to 3.0. Probably only commit 7c0dec9765d59552db4fea8888351ed100501d6d (fixing the chroot-nic-hotplug-issue). ATM the QA suites improvements are not on the backports list.

File: lib/hypervisor/hv_kvm/monitor.py
Comment (rbott) [reply lib/hypervisor/hv_kvm/monitor.py]: uh, thanks for spotting. This is actually wrong here ('romfile' is only relevant to the device_add structure, not to netdev_add)

File: lib/hypervisor/hv_kvm/__init__.py
Comment (rbott) [reply lib/hypervisor/hv_kvm/__init__.py]: fixed

**Comment by rbott** [comment]
Thanks @saschalucas. Sorry for the 80c and the leftover "romfile"-parameter. This has been fixed now. I'll wait for all tests to pass and will merge after that :-)

**Comment by rbott** [comment]
> Generally the ROM seems useless, as long as there is no intention to boot from PXE? Maybe Qemu (or Ganeti) should not load a ROM, unless explicitly requested (i.e. by `boot_order`)?

I am not sure if we should derive the need for a boot rom from the `boot_order`. I think qemu automatically falls through to network boot of there is no OS on disk present. But I might be wrong here.

**Pull request merged**
